### PR TITLE
Mgvalleys: Don't let cavegen place biome nodes everywhere

### PR DESCRIPTION
**Pull request by paramat**
Fix use of 'air_above' bool so that biome
nodes are only placed in tunnel floors
Minor code improvements
'Continue' when massive cave air is placed
//////////////////////////////

Same issue as #3885 but now fixed in mgvalleys.
Tested.

### CODE REVIEW

**Comment by kwolekr** [comment]
Just to be clear:  Everything in this commit is a style change except for the addition of the single `continue;` statement?
This should have been two separate commits IMHO.

But, :+1:

File: src/mapgen_valleys.cpp [hunk 1]
```diff
@@ -963,23 +961,24 @@ void MapgenValleys::generateCaves(s16 max_stone_y)
 
 			// River water is not set as ground content
 			// in the default game. This can produce strange results
-			// when a cave undercuts a river. However, that's not for
+			// when a tunnel undercuts a river. However, that's not for
 			// the mapgen to correct. Fix it in lua.
 
 			if (c == CONTENT_AIR) {
-				air_above = true;
+				continue;
```
Comment (kwolekr) [inline src/mapgen_valleys.cpp]: This `continue;` statement is ineffectual.  You should remove this entire if clause if no work is being performed.

Comment (paramat) [inline src/mapgen_valleys.cpp]: Agreed.

**Comment by paramat** [comment]
Hm i can see how useful it is for reviewers to keep code style edits seperate, sorry about that.

**Comment by paramat** [comment]
Updated and tested.

As requested removed unnecessary:
```
			if (c == CONTENT_AIR) {
				continue;
```
Also optimised structure from:
```
			} else if (d1 * d2 > 0.3f && ndef->get(c).is_ground_content) {
				............
			} else if (air_above && (c == biome->c_filler || c == biome->c_stone)) {
				............
				air_above = false;
				underground = true;
			} else if (c == biome->c_filler || c == biome->c_stone) {
				air_above = false;
				underground = true;
			} else {
				air_above = false;
			}
```
to:
```
			if (d1 * d2 > 0.3f && ndef->get(c).is_ground_content) {
				.............
			} else if (c == biome->c_filler || c == biome->c_stone) {
				if (tunnel_air_above) {
					..............
				}

				tunnel_air_above = false;
				underground = true;
			} else {
				tunnel_air_above = false;
			}
```

**Comment by paramat** [comment]
fe349d401d2d414f1ddc02c471c01d948bd89ee6

**Pull request closed without merging**